### PR TITLE
CLI: Warn when community addons are incompatible with Storybook 7

### PR DESCRIPTION
**Pull request by yannbf**
Relates to #20529

## What I did

☝️ See the linked issue for fore info

Added a banner to warn users in case they migrate and still have addons that are not compatible with Storybook 7 yet.

Dev (start command):
image

Build:
image

Automigration:
image

## How to test

1. Run a sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template react-vite/default-ts`
2. install an incompatible addon e.g. @storybook/addon-knobs
3. run start-storybook
4. get the banner

## Checklist

- [x] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [ ] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [x] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

**Comment by winkerVSbecks** [comment]
@yannbf I'd suggest rewording the message to:

> You’ll likely have to upgrade or disable these addons, and check the following Github issue for more information:

**Comment by joe-vaughan** [comment]
@yannbf -- fully agree with @winkerVSbecks

for suggested rewrite of the main issue page:

Many of Storybook's vast range of addons were made by developers from our community.

Years of changes are packed into Storybook 7. While all core Storybook addons have been updated with Storybook 7 compatibility, and many issues have been mitigated through automigrations, it is a massive task to ensure compatibility with all community addons, as many APIs have changed and may no longer be available.

To support developers as they migrate their addons to Storybook 7, we have created [an addon migration guide](https://chromatic-ui.notion.site/Addon-migration-for-Storybook-7-002f2f627c1d470298ca94894ab8a040) that outlines how to make your addons Storybook 7 compatible. We're keeping up-to-date records of the addons that are and aren't compatible with Storybook 7, and you can explore many of these in the lists below.

### Most popular community addons (by npm downloads)

[list]

### Community addons
_(Please share any addons with us for inclusion in this list)_

[list]
 
### FAQS

**I'm maintaining an addon. How do I make it compatible with Storybook 7?**

We highly recommend consulting [our abridged addon migration guide](https://chromatic-ui.notion.site/Addon-migration-for-Storybook-7-002f2f627c1d470298ca94894ab8a040). If you have any questions or encounter any issues, please reach out to Storybook maintainers in the #prerelease channel in the [Storybook Discord community](https://discord.gg/storybook).

**What do I do if an addon I'm using is incompatible with Storybook 7 but isn't on this list?**

To share incompatibility errors, please open an issue on the repo of the relevant addon, asking the author to add Storybook 7 compatibility, and share this page with them for reference.

It's also really useful for us to know which addons aren't yet compatible, so please consider adding a comment here, especially if the addon is popular and causes a critical failure.

Finally, if you have the resources, please consider helping migrate that addon yourself!

**Review by shilman — APPROVED** [review]
I think the current phrasing is good. We don't actually know the extent to which each addon is broken (or not) in 7.0, so we don't need to alarm users unnecessarily. As we get more visibility into which addons are broken and how badly, we can update the message/list accordingly. Merging!

**Comment by shilman** [comment]
@joe-vaughan thanks for the copy edits. have updated the issue text 👍
